interfaces/policy/helpers.go

```diff
@@ -41,7 +42,30 @@ func checkSnapType(snapType snap.Type, types []string) error {
 			return nil
 		}
 	}
-	return fmt.Errorf("snap type does not match")
+	return errors.New("snap type does not match")
+}
+
+var noIDMatchErr = errors.New("id does not match")
+
+func checkID(id string, ids []string, special map[string]string) error {
+	if len(ids) == 0 {
+		return nil
+	}
+	if id == "" { // unset never matches
+		return noIDMatchErr
```

Comment:
I think we'll need to inline here and provide a context in, so we can tell what does not match. As it is the errors for publisher id, snap id, etc are all the same, right?